JASP-Desktop/components/JASP/Controls/TableView.qml

```diff
@@ -249,4 +225,60 @@ JASPControl
 			bigBar:			false
 		}
 	}
+
+	Component
+	{
+		id: textFieldComponent
+		Item
+		{
+			TextField
+			{
+				id:						textInput
+				isBound:				false
+				anchors.verticalCenter: parent.verticalCenter
+				anchors.left:			parent.left
+				anchors.top:			parent.top
+				anchors.topMargin:		- 8 * preferencesModel.uiScale
+				fieldHeight:			parent.height + 16 * preferencesModel.uiScale
+				fieldWidth:				parent.width
+				visible:				true
+				useExternalBorder:		false
+				showBorder:				false
+				value:					itemText
+				useLastValidValue:		false
+				selectValueOnFocus:		true
+				validator:				tableView.validator
+				onPressed:				tableView.colSelected = columnIndex
+				onEditingFinished:		tableView.itemChanged(columnIndex, rowIndex, value)
+			}
+		}
+	}
+
+	Component
+	{
+		id: formulaFieldComponent
+		Item
+		{
+			FormulaField
+			{
+				id:						formlaInput
+				inputType:				columnIndex === 1 ? "formula" : "string"
```

Comment:
Could you not reuse this component entirely (poss. with an extra check here) also for `!tableView.modelType === "JAGSDataInputModel`?

FormulaField with `inputType == "formula"` is the same as TextField right?